internal/controllers/aggregation/slice.go

```diff
@@ -111,15 +111,17 @@ func (s *sliceController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 
 		if synthed := comp.Status.CurrentSynthesis.Synthesized; synthed != nil {
 			latency := now.Sub(synthed.Time)
-			logger.V(0).Info("composition was reconciled", "latency", latency.Milliseconds())
+			logger.V(0).Info("composition was reconciled", "latency", latency.Abs().Microseconds())
 		}
 	} else {
 		comp.Status.CurrentSynthesis.Reconciled = nil
 	}
 
 	err = s.client.Status().Update(ctx, comp)
 	if err != nil {
+		logger.V(0).Info(" Error in aggregating resource status into composition", "msg", err)
```

Comment:
I have a resistance to logging like this since it ends up with duplicate entries (the manager will log any returned errors as well).

Kind of related: I'd love to find some way to inject extra properties (compositionName, etc.) into the error logs emitted by the manager since right now they are sometimes lacking context.

Reply:
Let me know if this is what you were looking for , I added extra logging with , return ctrl.Result{}, fmt.Errorf("updating composition '%s' status: %w", comp.Name, err)

Reply:
That works for now. Ultimately I want to find a way to inject the fields from the controller itself, so they don't have to be added by every controller. But there are some perf concerns since it requires hitting the informer cache (IIRC), so I've been putting it off.

In any case, thanks for addressing this error - I'll make an issue for the fancy fields injection work.